src/thunderbots/software/ai/hl/stp/tactic/receiver_tactic.cpp

```diff
@@ -0,0 +1,172 @@
+    this->enemy_team    = updated_enemy_team;
+    this->ball          = updated_ball;
+    this->pass          = updated_pass;
+}
+
+double ReceiverTactic::calculateRobotCost(const Robot& robot, const World& world)
+{
+    // Prefer robots closer to the pass receive position
+    // We normalize with the total field length so that robots that are within the field
+    // have a cost less than 1
+    double cost =
+        (robot.position() - pass.receiverPoint()).len() / world.field().totalLength();
+    return std::clamp<double>(cost, 0, 1);
+}
+
+std::unique_ptr<Intent> ReceiverTactic::calculateNextIntent(
+    intent_coroutine::push_type& yield)
+{
+    MoveAction move_action = MoveAction(MoveAction::ROBOT_CLOSE_TO_DEST_THRESHOLD, true);
+    while (ball.lastUpdateTimestamp() < pass.startTime())
+    {
+        // We want the robot to move to the receiving position for the shot and also
+        // rotate to the correct orientation to face where the pass is coming from
+        yield(move_action.updateStateAndGetNextIntent(*robot, pass.receiverPoint(),
+                                                      pass.receiverOrientation(), 0));
+    }
+
+    // Check if we can shoot on the enemy goal from the receiver position
+    std::optional<std::pair<Point, Angle>> best_shot_opt =
+        calcBestShotOnEnemyGoal(field, friendly_team, enemy_team, *robot);
+
+    // Vector from the ball to the robot
+    Vector robot_to_ball = ball.position() - robot->position();
+
+    // The angle the robot will have to deflect the ball to shoot
+    Angle abs_deflection_angle;
+    // The percentage of open net the robot would shoot on
+    double net_percent_open;
+    if (best_shot_opt){
+        Vector robot_to_shot_target = best_shot_opt->first - robot->position();
+        abs_deflection_angle = (robot_to_ball.orientation() -
+         robot_to_shot_target.orientation())
+                .angleMod()
+                .abs();
+
+        Angle goal_angle = vertexAngle(field.friendlyGoalpostPos(), robot->position(), field.friendlyGoalpostNeg()).abs();
+        net_percent_open = best_shot_opt->second.toDegrees() / goal_angle.toDegrees();
+    }
+
+    // If we have a shot with a sufficiently large enough opening, and the deflection
+    // angle that is reasonable, we should one-touch kick the ball towards the enemy net
+    if (best_shot_opt &&
+         net_percent_open > MIN_SHOT_NET_PERCENT_OPEN &&
+         abs_deflection_angle < MAX_DEFLECTION_FOR_ONE_TOUCH_SHOT)
+    {
+        auto [best_shot_target, _] = *best_shot_opt;
+
+        // The angle between the ball velocity and a vector from the ball to the robot
+        Vector ball_velocity = ball.velocity();
+        robot_to_ball        = robot->position() - ball.position();
+        Angle ball_robot_angle =
+            (ball_velocity.orientation() - robot_to_ball.orientation()).angleMod();
+
+        while(ball_robot_angle.toDegrees() < 90 || ball.velocity().len() < 0.5)
```

Comment:
added:
```
        // Keep trying to shoot the ball while it's traveling roughly towards the robot
        // (or moving slowly because we can't be certain of the velocity vector if it is)
```